### ISSUE TITLE
Messed text-detection models output order

### ISSUE THREAD

**Issue by banderlog** [opencv/open_model_zoo#880]
In models description, since `text-detection-002` stated that:

> The net outputs two blobs. Refer to PixelLink and demos for details.
>+ [1x2x192x320] - logits related to text/no-text classification for each pixel.
>+ [1x16x192x320] - logits related to linkage between pixels and their neighbors.

But in  `text-detection-003` and  `text-detection-004` it is, actually, quite opposite.
Proof code:

```python
import cv2
import numpy as np

class PixelLinkTest():
    def __init__(self, xml_model_path: str):
        self._net = cv2.dnn.readNet(xml_model_path, xml_model_path[:-3] + 'bin')

    def detect(self, img: np.ndarray) -> None:
        """ GetPixelLink's outputs """
        self._img_shape = img.shape
        blob = cv2.dnn.blobFromImage(img, 1, (1280, 768))
        self._net.setInput(blob)
        out_layer_names = self._net.getUnconnectedOutLayersNames()
        self.pixels, self.links = self._net.forward(out_layer_names)
        
    def print_shapes(self) -> None:
        print(f"PIXELS: {self.pixels.shape}, LINKS: {self.links.shape}")


detector2 = PixelLinkTest('text-detection-0002.xml')
detector3 = PixelLinkTest('text-detection-0003.xml')
detector4 = PixelLinkTest('text-detection-0004.xml')

# random RGB image generation
img = (np.random.standard_normal((300, 300, 3)) * 255).astype(np.uint8)
detector2.detect(img)
detector3.detect(img)
detector4.detect(img)

detector2.print_shapes()
detector3.print_shapes()
detector4.print_shapes()
```
Output:
```
PIXELS: (1, 2, 192, 320), LINKS: (1, 16, 192, 320)
PIXELS: (1, 16, 192, 320), LINKS: (1, 2, 192, 320)
PIXELS: (1, 16, 192, 320), LINKS: (1, 2, 192, 320)
```

**Comment by jgespino** [comment]
Hi @banderlog 

I used the inference engine to read the output layer names and shapes (see output below). The layer names and shapes match as described in the documentation. However, I see the confusion as the documentation does not state the order of the output. The segm_logits layer is the text/no-text classification for each pixel and link_logits is the linkage between pixels and their neighbors.

text-detection-0002.xml (replaced by 0003 & 0004)
Output # 0 Name: pixel_cls/add_2
Output shape [1, 2, 192, 320]
Output # 1 Name: pixel_link/add_2
Output shape [1, 16, 192, 320]

text-detection-0003.xml
Output # 0 Name: model/link_logits_/add
Output shape [1, 16, 192, 320]
Output # 1 Name: model/segm_logits/add
Output shape [1, 2, 192, 320]

text-detection-0004.xml
Output # 0 Name: model/link_logits_/add
Output shape [1, 16, 192, 320]
Output # 1 Name: model/segm_logits/add
Output shape [1, 2, 192, 320]

Regards,
Jesus

**Comment by banderlog** [comment]
>  I see the confusion as the documentation does not state the order of the output. 

Exactly. Just switching strings in documentation would be enough, now it is confusing and somewhat misleading.

**Comment by IRDonch** [comment]
Well spotted. I'll fix it.

**Comment by IRDonch** [comment]
Fixed.
